### PR TITLE
Stategen: always invalidate hot cache state in StateByRootInitialSync

### PR DESCRIPTION
**Pull request by prestonvanloon**
**What type of PR is this?**

Bug fix

**What does this PR do? Why is it needed?**

This method claims to invalidate the cache, but there were some code paths where it didn't invalidate the cache. 

Using defer will ensure that all code paths will invalidate the cached value.

**Which issues(s) does this PR fix?**

None reported.

**Other notes for review**

Part of an investigation for corrupted hot state caches.

### CODE REVIEW

**Comment by prestonvanloon** [comment]
If there wasn't a cache entry there then it's a no-op. I don't think we need to add any check. The method could be renamed to `DeleteIfExists` to better represent what it's doing.

**Comment by prestonvanloon** [comment]
Actually, Delete returns true if something was deleted. Either way, no check is necessary imo

**Comment by nisdas** [comment]
> If there wasn't a cache entry there then it's a no-op. I don't think we need to add any check. The method could be renamed to DeleteIfExists to better represent what it's doing.

That would be true although I was thinking more of a simple regression test for it. We put in a state into the cache and then use that method and check that the cache does not have that particular state anymore.

**Comment by prestonvanloon** [comment]
Ah, a unit test! Yes, fair enough. I'm on it now.

File: beacon-chain/state/stategen/getter_test.go [hunk 1]
```diff
@@ -139,11 +139,14 @@ func TestStateByRootInitialSync_UseCache(t *testing.T) {
 	require.NoError(t, service.beaconDB.SaveStateSummary(ctx, &pb.StateSummary{Root: r[:]}))
 	service.hotStateCache.Put(r, beaconState)
 
-	loadedState, err := service.StateByRoot(ctx, r)
+	loadedState, err := service.StateByRootInitialSync(ctx, r)
```
Comment (prestonvanloon) [inline beacon-chain/state/stategen/getter_test.go:142]: Note: this test was called `TestStateByRootInitialSync_UseCache` but was calling a different method. Oops!

**Pull request merged**